code/modules/mob/new_player/new_player.dm

```diff
@@ -19,6 +19,11 @@
 	GLOB.mob_list += src
 	return INITIALIZE_HINT_NORMAL
 
+/mob/new_player/Destroy()
+	GLOB.player_list -= src
```

Comment:
Shouldn't this happen in `mob/Logout()`?

Reply:
Im not sure that gets called on mob transform, but will check

Reply:
it *should* happen each time a client leaves the mob and goes to another mob. This is why it's only ever removed in `mob/Logout()`. It would however be... something if this doesn't always happen

Reply:
Despite what the ref says, `/mob/Logout()` is never called unless a client is DCing. Swapping from mob-to-mob doesnt call any of the things in here, including but not limited to:
- TGUI window closing
- Generating logout logs
- Admin logout notifications
- Machine unsetting